app/src/main/java/ch/epfl/sweng/rps/LoginActivity.kt

```diff
@@ -7,42 +7,43 @@ import android.view.View
 import androidx.appcompat.app.AppCompatActivity
 import ch.epfl.sweng.rps.auth.FirebaseAuthenticator
 import ch.epfl.sweng.rps.models.User
-import ch.epfl.sweng.rps.storage.PrivateStorage
-import ch.epfl.sweng.rps.storage.Storage
+import ch.epfl.sweng.rps.persistance.Cache
+import ch.epfl.sweng.rps.persistance.PrivateStorage
 
 
 class LoginActivity : AppCompatActivity() {
 
     private var callback =
         { user: User ->
-            Log.d("Here", "HERASDASDA")
+
             launchMain(user)
         }
 
     private fun launchMain(user: User) {
         val intent = Intent(this, MainActivity::class.java)
         val b = Bundle()
-        store.writeBackUser(user)
         b.putString("email", user.email)
         b.putString("display_name", user.username)
         b.putString("uid", user.uid)
         b.putString("privacy", user.games_history_privacy.toString())
         intent.putExtra("User", b)
         startActivity(intent)
```

Comment:
Isn't there a prettier and scalabe way to do this ?

Reply:
yes thanks for catching that! It is not needed since the cache already does it, it came from old implementations